com.unity.sg2/Editor/GraphDeltaRegistry/Registry/ParametricTypeUtils.cs

```diff
@@ -0,0 +1,95 @@
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal static class ParametricTypeUtils
+    {
+        internal static string ParametricToHLSL(string name, int length, int height, float[] data)
+        {
+            string result = name + "(";
```

Comment:
I propose using a `StringBuilder` here instead of string concatenation.

```
            StringBuilder sb = new ('(');
            for (int i = 0; i < length * height; ++i)
            {
                sb.Append('{');
                sb.Append(data[i]);
                sb.Append('}');
                sb.Append(',');
            }
            sb.Length--;  // strip the last ',' before closing
            sb.Append(')');
            return sb.ToString();
```

Reply:
Done.